homeassistant/components/heos/media_player.py

```diff
@@ -86,6 +87,13 @@ def __init__(self, player):
 
     async def _heos_event(self, event):
         """Handle connection event."""
+        from pyheos import CommandError, const
+        if event == const.EVENT_CONNECTED:
+            try:
+                await self._player.refresh()
+            except (CommandError, asyncio.TimeoutError, ConnectionError) as ex:
```

Comment:
Will this have set the player as unavailable?

Reply:
Availability is set prior to the event being raised.  On connection, the player is available again. This is just for refreshing the latest playing media... it shouldn’t fail, but wanted to handle it gracefully if it does.